qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -39,22 +39,75 @@
 import qubesadmin.vm
 from qubesadmin.tools import xcffibhelpers
 
+# pylint: disable=wrong-import-position,wrong-import-order
+from Xlib import X, XK
+from Xlib.display import Display
+from Xlib.error import DisplayConnectionError
```

Comment:
Two issues here:
1. I believe nowadays it should be "xcffib" - "xcb" is preferred over "xlib" directly, and its python bindings are in python3-xcffib
2. You forgot to add dependency to the debian package, so qvm-start-daemon fails on all Debian debian tests.

Reply:
> 1. I believe nowadays it should be "xcffib" - "xcb" is preferred over "xlib" directly, and its python bindings are in python3-xcffib

I have to study xcffib documentation as it appears to be a drop-in replacement for xcb rather than xlib. It does not provide `alloc_named_color` and `string_to_keysym` functions (or I can not find it).

> 2\. You forgot to add dependency to the debian package, so qvm-start-daemon fails on all Debian debian tests.
OK. I will do it

Reply:
If there are no easy equivalents, then I guess it may stay with python3-xlib, but check first.

Reply:
https://stackoverflow.com/a/50539803
> (Except that it only shows keysym values, not keysym names, which you can find in /usr/include/X11/keysymdef.h or using Xlib's XKeysymToString(), and I don't think there exists an XCB port of that function, but you could write your own based on keysymdef.h.)

Reply:
> https://stackoverflow.com/a/50539803

As Geoffrey comments in the Stackoverflow, his suggested code does not show keysym names (only syms & values). And suggests `XKeysymToString` (which I am using it's python-xlib equivalent here).

For the keyboard, It might be possible to revert to [python3-xkbcommon](https://github.com/sde1000/python-xkbcommon) which is Python bindings for libxkbcommon using cffi. And what is used nowadays with [Wayland](https://wayland-book.com/seat/xkb.html) as well.

For the color, I have to look for alternatives

Reply:
@marmarek there are positive and negative things about `python3-xkbcommon`. I tested it and it works well. While it is the new modern approach and what is currently used with both X & Wayland, It is only available for Fedora 38 onward. Not Fedora 37. So it is time to decide if this patch is going to come only to Qubes R4.3 or R4.2 as well.

Reply:
For color, using `xcffib.xproto.xprotoExtension().AllocNamedColor()` works. But not as good as the Xlib's `alloc_named_color`. Since Xlib supports `#RGB`, `#RRGGBB`, `rgb:RRRRGGGGBBBB`, etc.

Reply:
> It is only available for Fedora 38 onward.

It also isn't available in Debian... So, I guess python3-xlib it is.

Reply:
> It also isn't available in Debian... So, I guess python3-xlib it is.

During the early stages of writing this PR, I used higher level GTK 4.0 for Keysym name validation. It is still possible. And the ugly approach of `try import xcffib except import xlib` is also possible. Still your decision.